pkg/database/mysql.go

```diff
@@ -0,0 +1,26 @@
+package database
+
+
+type MySQLDatabase struct {
+	DB *bun.DB
+}
+
+var (
+	once sync.Once
+	dbInstance *MySQLDatabase
+)
+
+func NewMySQLDatabase(conf *entities.Config) *MySQLDatabase {
+	once.Do(func() {
+		sqlDB, err := sql.Open("mysql", "")
+		if err != nil {
+			panic(err)
+		}
+		dbInstance := bun.NewDB(sqlDB, mysqldialect.New())
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Variable shadowing could cause issues</b></div>
<div id="fix">

The `dbInstance` variable is declared locally but should be assigned to the package level `dbInstance`. Consider using `:=` for local variables to avoid confusion.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
		dbInstance = &MySQLDatabase{DB: bun.NewDB(sqlDB, mysqldialect.New())}
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/13#issuecomment-2556324207>#a07780</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged